ui/ozone/platform/starboard/test/surface_factory_starboard_unittest.cc

```diff
@@ -0,0 +1,77 @@
+// Copyright 2024 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ui/ozone/platform/starboard/surface_factory_starboard.h"
+
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace ui {
+
+class SurfaceFactoryStarboardTest : public testing::Test {
+ public:
+  SurfaceFactoryStarboardTest() {}
+
+  void SetUp() override {
+    surface_factory_ = std::make_unique<SurfaceFactoryStarboard>();
+  }
+
+ protected:
+  SurfaceFactoryStarboard* surface_factory() const {
+    return surface_factory_.get();
+  }
+
+ private:
+  std::unique_ptr<SurfaceFactoryStarboard> surface_factory_;
+};
+
+TEST_F(SurfaceFactoryStarboardTest, GetGLOzoneSupportsEGLGLES2) {
+  auto gl_implementation = surface_factory()->GetGLOzone(
+      gl::GLImplementationParts(gl::kGLImplementationEGLGLES2));
+
+  EXPECT_NE(gl_implementation, nullptr);
+}
+
+TEST_F(SurfaceFactoryStarboardTest, GetGLOzoneSupportsEGLANGLE) {
+  auto gl_implementation = surface_factory()->GetGLOzone(
+      gl::GLImplementationParts(gl::kGLImplementationEGLANGLE));
+
+  EXPECT_NE(gl_implementation, nullptr);
+}
+
+TEST_F(SurfaceFactoryStarboardTest, GetGLOzoneDoesNotSupport) {
+  auto gl_implementation = surface_factory()->GetGLOzone(
+      gl::GLImplementationParts(gl::kGLImplementationNone));
+  EXPECT_EQ(gl_implementation, nullptr)
+      << "We don't expect to support kGLImplementationNone, but we returned a "
+         "real GL implementation.";
+
+  gl_implementation = surface_factory()->GetGLOzone(
+      gl::GLImplementationParts(gl::kGLImplementationMockGL));
+  EXPECT_EQ(gl_implementation, nullptr)
+      << "We don't expect to support kGLImplementationMockGL, but we returned "
+         "a real GL implementation.";
+
+  gl_implementation = surface_factory()->GetGLOzone(
+      gl::GLImplementationParts(gl::kGLImplementationStubGL));
+  EXPECT_EQ(gl_implementation, nullptr)
+      << "We don't expect to support kGLImplementationStubGL, but we returned "
+         "a real GL implementation.";
+
+  gl_implementation = surface_factory()->GetGLOzone(
+      gl::GLImplementationParts(gl::kGLImplementationDisabled));
+  EXPECT_EQ(gl_implementation, nullptr)
+      << "We don't expect to support kGLImplementationDisabled, but we "
+         "returned a real GL implementation.";
+}
```

Comment:
All of these test/cases are calling GetGLOzone() with a given implementation, then expecting a nullptr or not. This should be done with a parameterized test created with TEST_P, and using a list of pairs kGLImplementation + expected output. 

See the multiple examples in Chrome, e.g.
https://source.chromium.org/chromium/chromium/src/+/main:media/gpu/svc_layers_unittest.cc;l=304-305 and https://source.chromium.org/chromium/chromium/src/+/main:media/gpu/svc_layers_unittest.cc;l=492-496 .

Reply:
Done.